### ISSUE TITLE
Very small smileys

### ISSUE THREAD

**Issue by djizuss** [IBBoard/cawbird#414]
![Capture d’écran_2021-12-21_09-44-59](https://user-images.githubusercontent.com/8915292/146900401-ec4340e9-9d79-46c7-b721-49c1682ccd45.png)
Ubuntu 20.04 Xfce

**Comment by IBBoard** [comment]
That's weird. Looks like a font issue.

How do they render in the middle of text?

**Comment by djizuss** [comment]
Like this:
![image](https://user-images.githubusercontent.com/8915292/147859498-63fd6318-13b6-470d-9098-c941061be1ab.png)

**Comment by IBBoard** [comment]
I was going to say "but the one in the name renders correctly", but that appears to be a screenshot 🙂

Which desktop are you using? Which distro? And which version of Cawbird from which source?

I'm almost certain that this isn't our issue as we just hand off font rendering to GTK/Cairo and don't do anything special with emoji. If you can test with another GTK app (like `gtk3-widget-factory` - page 3 has an emoji button in the top-right) then we should be able to confirm.

**Comment by djizuss** [comment]
I use Xfce desktop within Ubuntu 20.04 (Xubuntu). Cawbird's version is 1.4.2 from the distribution but in... a snap ! Could it be the Snap ?
The Widget factory: no present in Ubuntu repository, no download possible on GWF SourceForge pages (is the project longer supported ?), even not installable from luigifab's ppa AWF: dependencies problem (libgdk-pixbuf-2 for gtk2 and 3 and libgtk-4 for gtk4) 
On Manjaro Xfce, CawBird 1.4.2 too from pacman -Syu (not AUR): no smileys's button.

I can event try to install CawBird from OpenSuze repository

**Comment by IBBoard** [comment]
I don't know how Snap interacts with fonts and things, but I have seen issues caused by sandboxed package formats. I prefer native packages because I find them easier to debug.

The widget factory is provided by `gtk3-devel` on my openSUSE system, but any GTK-based app should show the behaviour. If you're on Xubuntu rather than Kubuntu (I find it hard to tell the grey UIs apart!) then I think many of your apps will be GTK anyway.

If the smiley button isn't showing then Cawbird can't load your emoji data. Try running it from the command-line to see what errors. IIRC I need `gtk3-schema` (or `gtk4-schema`) which provides `/usr/share/glib-2.0/schemas/org.gtk.gtk4.Settings.EmojiChooser.gschema.xml` and then I can load the emoji data.

**Comment by djizuss** [comment]
GTK4 is not present yet on Xubuntu, so no file "org.gtk.gtk4.Settings.EmojiChooser.gschema.xml". I go for install it if possible.
Here is the command-line result:
`(cawbird:3784): cawbird-WARNING **: 09:20:05.913: ../src/src/CbUtils.c(call_done_cb): 0x55bc6a2e7520, 1.1/users/show.json Not Found
(cawbird:3784): cawbird-WARNING **: 09:20:05.913: Twitter.vala:239: Not Found
(cawbird:3784): cawbird-WARNING **: 09:20:06.283: ../src/src/CbUtils.c(call_done_cb): 0x55bc6a30ea20, 1.1/users/show.json Forbidden
(cawbird:3784): cawbird-WARNING **: 09:20:06.283: Twitter.vala:239: Forbidden`
I've also immediately install Valac but it didn't change anything.

Here is the command line on Manjaro, no smiley button, GTK4 is installed (but not GTK2 and 3), "org.gtk.gtk4.Settings.EmojiChooser.gschema.xml" is present:
`** (cawbird:1277): WARNING **: 10:26:47.181: Error loading plugin: libvoikko.so.1: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type
** (cawbird:1277): WARNING **: 10:26:47.183: Error loading plugin: libnuspell.so.5: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type
** (cawbird:1277): WARNING **: 10:26:47.201: Error loading plugin: libhspell.so.0: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type
** (cawbird:1277): WARNING **: 10:26:47.201: Error loading plugin: libaspell.so.15: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type
** (cawbird:1277): WARNING **: 10:26:47.223: Error loading plugin: libvoikko.so.1: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type
** (cawbird:1277): WARNING **: 10:26:47.224: Error loading plugin: libnuspell.so.5: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type
** (cawbird:1277): WARNING **: 10:26:47.226: Error loading plugin: libhspell.so.0: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type
** (cawbird:1277): WARNING **: 10:26:47.227: Error loading plugin: libaspell.so.15: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type`
"Cannot open shared object file: No such file or folder". But for me only languages packages ? Nothing related to GTK ? Can CawBird be modified by ArchLinux ?

**Comment by IBBoard** [comment]
> GTK4 is not present yet on Xubuntu, so no file "org.gtk.gtk4.Settings.EmojiChooser.gschema.xml". I go for install it if possible.
I assume the GTK3 package will install the GTK3 version and it should still work 🙂

All of those warnings look like they're related to the dictionary system.

I've just checked and we (obviously!) don't check for the emojis until you open "new tweet" dialog. When you do that you should see warnings about it failing to load files. I've also checked the source and it looks like we log everything except the final "could not load emojis" at debug level, so you'll actually need to run `G_MESSAGES_DEBUG=cawbird cawbird` to see what it's doing with trying to load emoji data. I see the following:

```
(cawbird:17192): cawbird-DEBUG: 20:10:29.708: Trying newer emoji data
(cawbird:17192): cawbird-DEBUG: 20:10:29.708: Using emoji/en.data resource
```

**Comment by djizuss** [comment]
Nothing about "emoji" in message debug on ubuntu. Complete output here: http://62.4.21.253/other/cawbird_debug
On Manjaro only gtk4. I see tomorrow for GTK3

**Comment by IBBoard** [comment]
Did you open the "Compose Tweet" dialog? Because there should be *something* about emoji, unless I got my logging very badly wrong! If you don't have the necessary emoji data then there should be lots of logging, but even if you do then there should be *something*!

**Comment by djizuss** [comment]
Here is what I did: I have open Cawbird with the debug command, I try to answer to a first tweet but was deleted, I try to answer to a second tweet (was a retweet) with 4 pictures, and I directly played with emojis (nothing was writing in the debug). Then I close the answer box keeping the message (some emojis), then I reopen and it wrotte line 192. Then I let the tweet selected and wait, then I close the debug.
Here is it: http://62.4.21.253/other/cawbird_debug_2

The reel problem is only to select the good emoji because they are too small. When selected and message sent it appear correctly on Tweeter.
![Capture d’écran_2022-01-08_19-46-50](https://user-images.githubusercontent.com/8915292/148656148-3015b7c4-edb2-4425-9cec-9a6e076908f0.png)

**Comment by IBBoard** [comment]
Twitter will be using both the browser's renderer and Twitter's images instead of Unicode characters, so it's not a realistic comparison. We need another GTK app that's rendering with your system fonts to see whether this happens across all GTK apps or somehow just happens in Cawbird.

According to your log, you're using an older version of GTK that has the `emoji.data` file:
```
(cawbird:29076): cawbird-DEBUG: 19:26:57.087: Using old emoji.data resource
(cawbird:29076): cawbird-DEBUG: 19:27:22.599: Using old emoji.data resource
```

**Comment by djizuss** [comment]
Here is something I see:
![Capture d’écran_2022-01-09_13-49-44](https://user-images.githubusercontent.com/8915292/148682894-978aa509-5951-4d32-9317-a304ca283ddb.png)
"I must be missing something about emoji support ... Clarification: WSL1, Ubuntu 20.04 + Terminator, font-noto-emoji + font-noto-mono. "NTFS" mount point but also the base LXFS. You know why ?"
"I haven't had a lot of "pull" on this tweet. But I found the solution: Ubuntu font apparently hasn't been updated for a while. I switched to Noto Mono (and noto emoji next to it, just in case), and I got the emoji!"
Do not work for me. Noto do not change anything
